src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl_ModelComputation.cs

```diff
@@ -0,0 +1,208 @@
+            var sortOption = SortOption;
+
+            // Switch to the threadpool to filter and sort the data model.
+            await TaskScheduler.Default;
+
+            var updatedDocumentSymbolData = model.DocumentSymbolData;
+
+            if (!string.IsNullOrWhiteSpace(searchQuery))
+                updatedDocumentSymbolData = DocumentOutlineHelper.SearchDocumentSymbolData(updatedDocumentSymbolData, searchQuery, cancellationToken);
+
+            updatedDocumentSymbolData = DocumentOutlineHelper.SortDocumentSymbolData(updatedDocumentSymbolData, sortOption, cancellationToken);
+
+            StartHighlightExpandAndPresentItemsTask(ExpansionOption.NoChange);
+
+            return new DocumentSymbolDataModel(updatedDocumentSymbolData, model.OriginalSnapshot);
+        }
+
+        /// <summary>
+        /// Starts a new task to highlight the symbol node corresponding to the current caret position in the editor, expand/collapse
+        /// nodes, and update the UI.
+        /// </summary>
+        private void StartHighlightExpandAndPresentItemsTask(ExpansionOption expansionOption)
+        {
+            _highlightExpandAndPresentItemsQueue.AddWork(expansionOption);
+        }
+
+        /// <summary>
+        /// Highlights the symbol node corresponding to the current caret position in the editor, expands/collapses nodes, then updates the UI.
+        /// </summary>
+        private async ValueTask HighlightExpandAndPresentItemsAsync(ImmutableSegmentedList<ExpansionOption> expansionOption, CancellationToken cancellationToken)
+        {
+            var model = await _filterAndSortDataModelQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(false);
+            if (model is null)
+                return;
+
+            // Switch to the UI thread to get the current caret point and latest active text view then create the UI model.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var activeTextView = GetLastActiveIWpfTextView();
+            if (activeTextView is null)
+                return;
+
+            var caretPoint = activeTextView.GetCaretPoint(activeTextView.TextBuffer);
+            if (!caretPoint.HasValue)
+                return;
+
+            var documentSymbolUIItems = DocumentOutlineHelper.GetDocumentSymbolUIItems(model.DocumentSymbolData, _threadingContext);
+
+            // Switch to the threadpool to determine which node to select (if applicable).
+            await TaskScheduler.Default;
+
+            var symbolToSelect = DocumentOutlineHelper.GetDocumentNodeToSelect(documentSymbolUIItems, model.OriginalSnapshot, caretPoint.Value);
+
+            // Switch to the UI thread to update the view.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            // Expand/collapse nodes based on the given Expansion Option.
+            var expansion = expansionOption.First();
```

Comment:
we actually want the *last* expansion option here.  consider if the user expands/collapses.  you'll have those two items in teh queue, but you'll lose the action they did later.  

because all later actions override previous actions, 'last' is always teh correct thing to do.